pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/ArrayGenerateTransformFunction.java

```diff
@@ -0,0 +1,223 @@
+        double endDouble = Double.parseDouble(literalContexts.get(1).getLiteral().getStringValue());
+        double incDouble = Double.parseDouble(literalContexts.get(2).getLiteral().getStringValue());
+        size = (int) ((endDouble - startDouble) / incDouble +1);
+        _doubleArrayLiteral = new double[size];
+        for (int i = 0 ; i < size; i ++, startDouble += incDouble) {
+          _doubleArrayLiteral[i] = startDouble;
+      }
+        _intArrayLiteral = null;
+        _longArrayLiteral = null;
+        _floatArrayLiteral = null;
+        break;
+      default:
+        throw new IllegalStateException(
+            "Illegal data type for ArrayGenerateTransformFunction: " + _dataType + ", literal contexts: "
+            + Arrays.toString(literalContexts.toArray()));
+    }
+  }
+
+  public int[] getIntArrayLiteral() {
+    return _intArrayLiteral;
+  }
+
+  public long[] getLongArrayLiteral() {
+    return _longArrayLiteral;
+  }
+
+  public float[] getFloatArrayLiteral() {
+    return _floatArrayLiteral;
+  }
+
+  public double[] getDoubleArrayLiteral() {
+    return _doubleArrayLiteral;
+  }
+
+  @Override
+  public String getName() {
+    return FUNCTION_NAME;
+  }
+
+  @Override
+  public void init(List<TransformFunction> arguments, Map<String, ColumnContext> columnContextMap) {
+
+  }
+
+  @Override
+  public TransformResultMetadata getResultMetadata() {
+    return new TransformResultMetadata(_dataType, false, false);
+  }
+
+  @Nullable
+  @Override
+  public Dictionary getDictionary() {
+    return null;
+  }
+
+  @Override
+  public int[] transformToDictIdsSV(ValueBlock valueBlock) {
+    return new UnsupportedOperationException();
+  }
+
+  @Override
+  public int[][] transformToDictIdsMV(ValueBlock valueBlock) {
+    return new UnsupportedOperationException();
+  }
+
+  @Override
+  public int[] transformToIntValuesSV(ValueBlock valueBlock) {
+    return new UnsupportedOperationException();
+  }
+
+  @Override
+  public long[] transformToLongValuesSV(ValueBlock valueBlock) {
+    return new UnsupportedOperationException();
+  }
+
+  @Override
+  public float[] transformToFloatValuesSV(ValueBlock valueBlock) {
+    return new UnsupportedOperationException();
+  }
+
+  @Override
+  public double[] transformToDoubleValuesSV(ValueBlock valueBlock) {
+    return new UnsupportedOperationException();
+  }
+
+  @Override
+  public BigDecimal[] transformToBigDecimalValuesSV(ValueBlock valueBlock) {
+    return new UnsupportedOperationException();
+  }
+
+  @Override
+  public String[] transformToStringValuesSV(ValueBlock valueBlock) {
+    return new UnsupportedOperationException();
+  }
+
+  @Override
+  public byte[][] transformToBytesValuesSV(ValueBlock valueBlock) {
+    return new UnsupportedOperationException();
+  }
+
+  @Override
+  public int[][] transformToIntValuesMV(ValueBlock valueBlock) {
+    return new int[0][];
```

Comment:
Here we should convert array literal and cache it.
Check how `ArrayLiteralTransformFunction` did here.

Reply:
Yes. Its on my to do list too. No worries :smile: